File: backend/app/Models/Quiz.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Quiz extends Model
{
    use HasFactory;


    protected $fillable = [
    'title',
    'description',
    'start_datetime',
    'end_datetime',
    'course_id',
    'class_room_id',
    'status',
    'user_id',
    'is_active',
];

protected $casts = [

    'start_datetime'=>'datetime',
    'end_datetime'=>'datetime',
    'is_active'=>'boolean',

];



    /**
     * Quiz belongs to lecturer
     */
    public function user()
    {
        return $this->belongsTo(User::class);
    }



    /**
     * Quiz has many questions
     */
    public function questions()
    {
        return $this->hasMany(Question::class);
    }

    



    /**
     * Quiz has many student attempts
     */
    public function attempts()
    {
        return $this->hasMany(Attempt::class);
    }

    public function course()
{
    return $this->belongsTo(Course::class);
}

public function classRoom()
{
    return $this->belongsTo(ClassRoom::class);
}

}

File: src/main/java/models/Quiz.java
package com.educonnect.models;


public class Quiz {

    private int id;
    private String title;
    private String description;
    private int duration;
    private String status;


    public Quiz(
            int id,
            String title,
            String description,
            int duration,
            String status
    ){

        this.id = id;
        this.title = title;
        this.description = description;
        this.duration = duration;
        this.status = status;

    }



    public int getId(){

        return id;

    }


    public String getTitle(){

        return title;

    }


    public String getDescription(){

        return description;

    }


    public int getDuration(){

        return duration;

    }


    public String getStatus(){

        return status;

    }


}